src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -695,13 +697,19 @@ public TypesByNamespaceSortComparer(StringComparer nameComparer)
                 _nameComparer = nameComparer;
             }
 
-            public int Compare(IGrouping<string, TypeDefinitionHandle> left, IGrouping<string, TypeDefinitionHandle> right)
+            public int Compare(IGrouping<string, TypeDefinitionHandle>? left, IGrouping<string, TypeDefinitionHandle>? right)
             {
                 if (left == right)
                 {
                     return 0;
                 }
 
+                if (left is null)
+                {
+                    return -1;
+                }
+
+                Debug.Assert(right is not null);
```

Comment:
Why cannot `right` be `null` here? I expected to see

```cs
if (right is null)
{
    return 1;
}
```

Reply:
Ugg ... was too asleep writing this method.